File: src/shared/Foundation.Extension.Domain/Models/PermissionApplications/PermissionApplicationCategory.cs
using System;
using System.Collections.Generic;

namespace Foundation.Extension.Domain.Models
{
    public class PermissionApplicationCategory: ITranslatable<TranslationPermissionApplicationCategory>
    {
        public string Label { get; set; }
        public string Prefix { get; set; }
        public List<TranslationPermissionApplicationCategory> Translations { get; set; }
    }

    public class TranslationPermissionApplicationCategory: ITranslation
    {
        public string LanguageCode { get; set; }
        public string Label { get; set; }
    }
}

File: src/app/Foundation.Extension.Admin-UI/domain/models/permissionApplications/permissionApplicationCategory.ts
export class PermissionApplicationCategory {
    label: string;
    prefix: string;

    constructor(params: PermissionApplicationCategoryDTO) {
        this.label = params.label;
        this.prefix = params.prefix;
    }
}

export interface PermissionApplicationCategoryDTO {
    label: string;
    prefix: string;
}